rados/ioctx.go

```diff
@@ -122,6 +149,45 @@ func (ioctx *IOContext) Append(oid string, data []byte) error {
 	return GetRadosError(int(ret))
 }
 
+// Executes read operation with librados operation flags
+func (ioctx *IOContext) ReadOperate(oid string, data []byte,
+	offset uint64) (int, error) {
+	if len(data) == 0 {
+		return 0, nil
+	}
+
+	c_oid := C.CString(oid)
+	defer C.free(unsafe.Pointer(c_oid))
+
+	r_op := C.rados_create_read_op()
+	if unsafe.Pointer(r_op) == nil {
+		return 0, GetRadosError(C.EIO)
+	}
+	defer C.rados_release_read_op(r_op)
+
+	var bytesRead uint64
+	var preval int
+	C.rados_read_op_read(r_op,
```

Comment:
is calling rados_read() same as calling this api with flags as 0?
I ask this as can we have Read() api call this function (or both of these functions call common internal function). That will avoid some duplication of code.

Reply:
Implementations look different here - https://github.com/ceph/ceph/blob/7720ef5dbdc171152f492f15ed39a82bf3a86631/src/librados/librados.cc. I thought of this as different APIs. rados_read_op is basically used for executing read operation with some context atomically. It could also set extent attributes as far as APIs is concerned. Keeping it different would help, I will add a comment here.